### DISCUSSION TITLE
[Feature request] Task Priority and Sub-Tasks

### DISCUSSION THREAD

**Discussion by JPL-DSCS** [standardnotes/forum#2636] [Suggestions]
**Is your feature request related to a problem? Please describe.**
Using the Task Editor Extension in iOS & MacOS apps I'd like to be able to prioritize and group my tasks other than by simply ordering them.

**Describe the solution you'd like**
1) _Task Priority_ : It would be great if we could give some sort of 'priority' (with several priority levels ideally) tag to any one task that would be immediately visible (with an icon, color or sth else), a bit like what Apple does with its Reminders App (screenshot below from their Desktop App) :

<img width="278" alt="image" src="https://user-images.githubusercontent.com/36039972/107712293-303b8280-6c97-11eb-8997-95acf3f694c5.png">

Even better, would be that the user be able to order its tasks according to their priority levels, so the urgent ones would all be grouped at the top, the non urgent ones stacked at the bottom of the list and so on...

2) _Sub-Tasks_: Additionally, it'd be awesome if we could decompose a task into sub-tasks, like nested fields with their own checkboxes to check so as to complete the main parent task.
I'm thinking of something that would look roughly like that (taken from Notion Desktop App) : 

<img width="219" alt="image" src="https://user-images.githubusercontent.com/36039972/107714246-ee144000-6c9a-11eb-9fe6-6f073b6756b0.png">

**Describe alternatives you've considered**
1) _Task Priority_ : Alternatively, a less fancy and precise feature could be just using a number-based ordering of tasks, ideally sorted ASC starting at 1 (most urgent)

2) _Sub-Tasks_: Alternatively, a less fancy and precise feature than the parent / child nested fields could be to implement "fake sub-tasks" by just having the ability to indent a task by a tab so that one could write indented tasks below one task to make them look like related sub-tasks 

**Additional context**
I think these proposed changes would make the Task Editor Extension a more granular and powerful tool in Standard Notes

*Apologies, I should have probably started two issues for two separate features...

**Comment by JaspalSuri** [comment]
Hi @JPL-DSCS,

Thanks for the feedback! I'll add your +1 to add the ability to create subtasks, and put in a request to assign the task's priority for review.

**Comment by JPL-DSCS** [comment]
Hi @JaspalSuri,
Thanks a lot for taking into consideration my suggestions 🙂 
I didn't know the subtasks feature had already been brought up, though I did a search of the open issues before posting... 🤔

Where can I follow that one and the task's priority feature request's advancements / outcome? Sorry for the newbie questions!

**Comment by JaspalSuri** [comment]
> I didn't know the subtasks feature had already been brought up, though I did a search of the open issues before posting... 🤔

No worries. There are at least some from reddit and a couple from Slack, so that's probably why. In any case, it's fine. 🙂 

> Where can I follow that one and the task's priority feature request's advancements / outcome? Sorry for the newbie questions!

We currently don't have a public timeline or roadmap, but the dev team is working on session management, simplifying privileges (regarding restricting access to parts of the app), sharing text to the mobile apps, natively uploading files and viewing images within the apps, and providing more batch management options. 🙂

**Comment by JPL-DSCS** [comment]
Got it! Thanks a lot for all the info 🙂

**Comment by Swivelgames** [comment]
Add me as a plus one as well!

These are big items for me, with the Sub-Tasks feature being a crucially important one for me personally. This feature is actually, honestly, one of the main reasons I haven't switched over from Google Keep. With Google Keep, it's super easy to indent a task to turn it into a sub-task. That's literally all I need to be able to switch to this for my daily, well, everything.

**Comment by siopaosmuggler** [comment]
Commenting to make sure this stays alive. I used a separate program, ToDoList by Abstractspoon, instead of StandardNotes' Task Editor, because ToDoList is more robust. I don't expect StandardNotes to implement all of these features (which include due dates, time tracking, infinite nesting, priority setting, and individual statuses, just to name a few), but the inability to add sub-tasks limits my interest in using StandardNotes' task editor for anything more complicated than general lists, like groceries or shows I'd like to watch some day.

**Comment by myreli** [comment]
You might want to experiment the new [Super](https://www.youtube.com/watch?v=bkp6nrGqjUM&t=370s) editor, which offers this ability: 

<img width="786" alt="image" src="https://user-images.githubusercontent.com/17554234/218856025-2304501e-ad79-4829-93a8-4e409a686666.png">
